src/node-jspm-jasmine.js

```diff
@@ -141,17 +179,79 @@ export function runTests(opts, errCallback = function() {}) {
 							importTheseTestFiles();
 						}
 					})
-					.catch(errCallback);
+					.catch((ex) => {
+						if (opts.coverage && opts.coverage.tempDirectory) {
+							// remove temporary directory
+							rimraf.sync(opts.coverage.tempDirectory)
+						}
+						errCallback(ex)
+					});
 				});
 			});
 		}
 	} catch(ex) {
-		removeTempFiles();
+		if (opts.coverage && opts.coverage.tempDirectory) {
+			// remove temporary directory
+			rimraf.sync(opts.coverage.tempDirectory)
+		}
 		errCallback(new Error(`Jasmine or Jspm may not be properly configured -- '${ex.toString()}'`));
 	}
 }
 
 function importTestFiles(SystemJS, jasmine, specDir, specFiles, coverage, errCallback) {
+
+	if (coverage) {
+		// jasmine will exit the process on its own, and since there doesn't appear
+		// to be a "onExit" hook, we're forced to intercept node's exit function and
+		// use as a "all done" jasmine hook.
+		// jasmine does offer a "addReporter" option, which contains a "jasmineDone"
+		// hook, however, it has no mechanism for waiting on async operations and
+		// async is needed for importing modules.
+		// Also, we should provide a "onExit" hook for the "programmatic" version,
+		// otherwise jasmine will abruptly kill the current running script.
+		const processExit = process.exit;
+		process.exit = function () {
```

Comment:
@joeldenning =O that's terrible. It appears that user termination maps to plain ol' "process.exit", which is intercepted at that point.

Regarding (1) and (2), both valid solutions, thank you for the proposition. I had a variation of (2) that I was preferring, but to our happiness, I just happen to be an idiot, and there is in fact an "onComplete" hook for Jasmine.[ It was there all along](http://jasmine.github.io/2.3/node.html#section-Custom_onComplete) and looking at the source, it does exactly what we want. Unlike the "onComplete" or "jasmineDone" reporter hooks, it delegates the exit to us. 

Btw, not sure if you recall, but  [this was the issue I was complaining about before](https://github.com/CanopyTax/node-jspm-jasmine/pull/7#issuecomment-210862594). It's been finally reasoned. Lastly, I would have resisted (1) a bit for the sake of avoiding multiple imports of modules, although I don't think that would have messed up the coverage reports.

Updating pr in a bit.